### PR TITLE
bcftools: import from homebrew/science

### PR DESCRIPTION
**Pull request by sjackman**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

### CODE REVIEW

File: Formula/bcftools.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  depends_on "xz"
+
+  def install
+    # Bug reported upstream and fixed in the next release.
+    # https://github.com/samtools/bcftools/issues/684
+    inreplace "Makefile",
+      "PLUGIN_FLAGS = -bundle -bundle_loader bcftools",
+      "PLUGIN_FLAGS = -bundle -bundle_loader bcftools -Wl,-undefined,dynamic_lookup"
+
+    system "./configure",
```
Comment (ilovezfs) [inline Formula/bcftools.rb]: same formatting thing

File: Formula/bcftools.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    system "make"
+
+    system "make", "install"
+
+    # For brew test bcftools
+    pkgshare.install "test/query.vcf"
+  end
+
+  test do
+    assert_match "number of SNPs:\t3", shell_output("#{bin}/bcftools stats #{pkgshare}/query.vcf")
```
Comment (ilovezfs) [inline Formula/bcftools.rb]:
```
    output = shell_output("#{bin}/bcftools stats #{pkgshare}/query.vcf")
    assert_match "number of SNPs:\t3", output
```

File: Formula/bcftools.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      "--with-htslib=#{Formula["htslib"].opt_prefix}",
+      "--enable-libgsl"
+
+    # Fix install: cannot stat ‘bcftools’: No such file or directory
+    # Reported upstream: https://github.com/samtools/bcftools/issues/727
+    system "make"
+
+    system "make", "install"
+
+    # For brew test bcftools
```
Comment (ilovezfs) [inline Formula/bcftools.rb]: this is self-evident I think

File: Formula/bcftools.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  homepage "http://www.htslib.org/"
+  url "https://github.com/samtools/bcftools/releases/download/1.6/bcftools-1.6.tar.bz2"
+  sha256 "293010736b076cf684d2873928924fcc3d2c231a091084c2ac23a8045c7df982"
+
+  depends_on "gsl"
+  depends_on "htslib"
+  depends_on "xz"
+
+  def install
+    # Bug reported upstream and fixed in the next release.
```
Comment (ilovezfs) [inline Formula/bcftools.rb]: like to have the date too

File: Formula/bcftools.rb [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+      "PLUGIN_FLAGS = -bundle -bundle_loader bcftools",
+      "PLUGIN_FLAGS = -bundle -bundle_loader bcftools -Wl,-undefined,dynamic_lookup"
+
+    system "./configure",
+      "--prefix=#{prefix}",
+      "--with-htslib=#{Formula["htslib"].opt_prefix}",
+      "--enable-libgsl"
+
+    # Fix install: cannot stat ‘bcftools’: No such file or directory
+    # Reported upstream: https://github.com/samtools/bcftools/issues/727
```
Comment (ilovezfs) [inline Formula/bcftools.rb]: ditto

**Comment by sjackman** [comment]
I've addressed your comments.

**Comment by ilovezfs** [comment]
@sjackman what's with the O0 in https://github.com/samtools/bcftools/commit/2384efbad7f55fcede759db2c7344aa2fd4e4c10

**Comment by sjackman** [comment]
Unintentional change. See https://github.com/samtools/bcftools/commit/b7b502e908cad70e613b03d5c64bbcea48a2df8d
> Revert unintended change from -O2 to -O0

**Comment by ilovezfs** [comment]
cool

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
Thanks again @sjackman! :wrench:

**Comment by sjackman** [comment]
Thanks for merging!

**Pull request closed without merging**